### ISSUE TITLE
Fade into the background?

### ISSUE THREAD

**Issue by KennyGoers** [Faltenreich/SkeletonLayout#7]
I noticed that depending on colors chosen and time spent waiting, showOriginal() can be quite jarring. Thoughts on how hard or easy it would be to add a fade in or out animation?

**Comment by Faltenreich** [comment]
A fade in is a very good idea and something I have thought about since the creation of this library. 

Currently this is not supported but could be achieved by altering the View toggling behind showOriginal() and showSkeleton().

Things are different for the RecyclerView, since I am creating a separate adapter for the skeleton that replaces the original one when calling showSkeleton() in order to keep the library as unintrusive as possible. 

I will think about a solution and of course you are free to create a pull request which I would gladly review if you are in a hurry.

**Comment by KennyGoers** [comment]
Been reviewing the code and looking at adding this for both the mask types. In the process I've noticed that in the current implementation set the VISIBLE/INVISIBLE state of all the views within the parent. Am I reading that correctly? If so it causes havoc with views that are initially set to anything other than VISIBLE. If this is the case (and I think it is) more needs to be checked in that pass and original visibility state maintained. Agree?

**Comment by Faltenreich** [comment]
You are absolutely right. The original view state should be cached in order to maintain the original visibilities. I created a new issue for this bug which may be fixed by me in the near future or by someone other if it is of high priority: https://github.com/Faltenreich/SkeletonLayout/issues/9

**Comment by mhettig** [comment]
A workaround can be achieved with:
```kotlin
// Example callback that hides skeleton
private fun onDataLoaded() {
    TransitionManager.beginDelayedTransition(skeleton)
    skeleton.showOriginal()
}
```
(Kotlin demo code from start page)
